### PR TITLE
Show insulin activity in BG graph

### PR DESCRIPTION
**Pull request by Paulusma**
Displays insulin activity (history & prediction) in BG graph as a thin white line.

I think a fixed scale for IA is required to allow a user to compare current IA with that from previous situations. Helps gaining insight in IA trends vs BG trends.

Final optionality & location of IA (could also be included in secondary graph) should be discussed including current performance overhead due to duplicate calls to iobCobCalculatorPlugin.calculateFromTreatmentsAndTempsSynchronized (addActivity is a copy of addIob)

### CODE REVIEW

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java [hunk 1]
```diff
@@ -1523,6 +1523,9 @@ else if (Config.NSCLIENT)
             // set manual x bounds to have nice steps
             graphData.formatAxis(fromTime, endTime);
 
+            // insulin activity
+            graphData.addActivity(fromTime, endTime, graphData.maxY);
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java]: make it conditional like `if (SP.getBoolean("showactivity", false)) .....`

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/graphData/GraphData.java [hunk 2]
```diff
@@ -344,6 +345,35 @@ private double getNearestBg(long date) {
...
+
+            actArray.add(new ScaledDataPoint(time, act, actScale));
+            lastAct = act;
+        }
+
+        ScaledDataPoint[] actData = new ScaledDataPoint[actArray.size()];
+        actData = actArray.toArray(actData);
+        actSeries = new FixedLineGraphSeries<>(actData);
+        actSeries.setDrawBackground(false);
+        actSeries.setColor(MainApp.gc(R.color.mdtp_white));
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/overview/graphData/GraphData.java]: define new color indentifier (can be white)

**Comment by Paulusma** [comment]
I changed the code so the graph context menu groups primary and secondary graph items. I also created a preference to scale the insulin activity in the graph

**Comment by Paulusma** [comment]
Changed the linestyle for future activity to dashed

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java (hunk 1 above)
Comment (Paulusma) [reply app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java]: done

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/graphData/GraphData.java (hunk 2 above)
Comment (Paulusma) [reply app/src/main/java/info/nightscout/androidaps/plugins/general/overview/graphData/GraphData.java]: done

**Comment by Paulusma** [comment]
Unintended changes in this branch has created a mess (sorry, Git * branches is new to me). I'll create a new branch.

**Pull request closed without merging**